app-web/src/components/Home/ResourcePreview.js

```diff
@@ -38,43 +40,179 @@ export const ResourceContainer = styled.div`
+export const ResourcePreview = ({ title, link, resources, filters, amountToShow, seeMore }) => {
+  let [showCount, updateCount] = useState(amountToShow);
+  let [seeMoreResults, updateSeeMore] = useState(seeMore);
+  let [resourcesToShow, updateResources] = useState(resources);
+  let [activeFilter, updateFilter] = useState('All');
+
+  //sets the amount of resources to show, allowing users to 'see more' if its appropriate
+  const setCount = () => {
+    //show 6 more results
+    updateCount(showCount + 6);
+    updateSeeMore(true);
+    if (showCount >= resourcesToShow.length) {
+      //hide the 'see more results' when there isnt more to show
+      updateSeeMore(false);
+    }
+  };
+
+  //This filters what results we are showing based on the given filter coming from user interaction with the ResourcePills
+  const resourceFilter = filter => {
+    //filter the results based on given filter, update the resources and active filter
+    let filteredResources = resources.filter(
+      resource => resource.fields.resourceType === filter.name,
+    );
+    updateResources(filteredResources);
+    updateFilter(filter.name);
+    //reset the amount of resources to show
+    updateCount(amountToShow);
+  };
+
+  //when 'All Results' pill is toggled, reset our variables to initial states and show all results
+  const showAllResults = () => {
+    updateResources(resources);
+    updateFilter('All');
+    updateCount(amountToShow);
+    updateSeeMore(seeMore);
+  };
+
+  let pills;
+  //Filters will be mapped into pills displaying result count for that particular resourcetype
+  //These pills are interactive and filter results when clicked
+  if (filters) {
+    //No people resource type Icon rn
+    pills = filters.map(filter => {
+      if (filter.name !== 'People') {
+        //formats the text correctly for different cases
+        let iconLabel =
+          filter.counter > 1 || filter.counter === 0
+            ? `${filter.counter} Results`
+            : `${filter.counter} Result`;
+        //adds informative info for the behavior of the ResourcePills their current state
+        let iconInfo =
+          filter.name === activeFilter
+            ? 'Click to view all results again'
+            : `Click to view only ${filter.name} results`;
+
+        return (
+          <ResourcePill
+            resourceType={filter.name}
+            label={iconLabel}
+            variant="filled"
+            deletable={false}
+            key={filter.name}
+            css={filter.name === activeFilter ? toggled : ''}
+            onClick={filter.counter === 0 ? undefined : () => resourceFilter(filter)}
+            title={iconInfo}
           />
-        </CardWrapper>
-      ))}
-    </ResourceContainer>
-    <LinkContainer>
-      <ChevronLink to={link.to}>{link.text}</ChevronLink>
-    </LinkContainer>
-  </Container>
-);
+        );
+      }
+      return '';
+    });
+    //Add final pill to the start of the list
+    pills.unshift(
```

Comment:
hmm i'd avoid using unshift. I think it won't cause any issues here but it is an object mutating method. Since we are typically using concat/filter to add and remove elements. Let's stick to that? What do you think?